amd_openvx_extensions/amd_rpp/source/tensor/TensorAddTensor.cpp

```diff
@@ -0,0 +1,237 @@
+    STATUS_ERROR_CHECK(vxSetMetaFormatAttribute(metas[2], VX_TENSOR_FIXED_POINT_POSITION, &tensor_fixed_point_position, sizeof(tensor_fixed_point_position)));
+    return status;
+}
+
+static vx_status VX_CALLBACK processTensorAddTensor(vx_node node, const vx_reference *parameters, vx_uint32 num) {
+    vx_status status = VX_SUCCESS;
+    TensorAddTensorLocalData *data = NULL;
+    STATUS_ERROR_CHECK(vxQueryNode(node, VX_NODE_LOCAL_DATA_PTR, &data, sizeof(data)));
+
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_OPENCL
+        return VX_ERROR_NOT_IMPLEMENTED;
+#elif ENABLE_HIP
+        return VX_ERROR_NOT_IMPLEMENTED;
+#endif
+    } else if (data->deviceType == AGO_TARGET_AFFINITY_CPU) {
+        refreshTensorAddTensor(node, parameters, num, data);
+        if (data->inTensorType == vx_type_e::VX_TYPE_FLOAT32 && data->outTensorType == vx_type_e::VX_TYPE_FLOAT32) {
+            // data->pSrcRoi = (RpptROI *)data->roi_tensor_ptr_src;
+            size_t nStride = data->inputTensorDims1[1] * data->inputTensorDims1[2];
+            for (uint i = 0; i < data->nbatchSize; i++) {
+                float *src1Temp = static_cast<float *>(data->pSrc1) + i * nStride;
+                float *src2Temp = static_cast<float *>(data->pSrc2) + i;
+                float *dstTemp = static_cast<float *>(data->pDst) + i * nStride;
+                uint height = data->pSrcRoi[i].xywhROI.xy.y;
+                uint width = data->pSrcRoi[i].xywhROI.xy.x;
+                uint alignedWidth = (width / 8) * 8;
+                float additionFactor = src2Temp[0];
+                __m256 pSrc2 = _mm256_set1_ps(additionFactor);
+                for (uint row = 0; row < height; row++) {
+                    float *srcPtr1Row = src1Temp + row * data->inputTensorDims1[1];
+                    float *dstPtrRow = dstTemp + row * data->inputTensorDims1[1];
+                    uint vectorLoopCount = 0;
+                    for (; vectorLoopCount < alignedWidth; vectorLoopCount += 8) {
+                        __m256 pSrc1 = _mm256_loadu_ps(srcPtr1Row);
+                        __m256 pDst = _mm256_add_ps(pSrc1, pSrc2);
+                        _mm256_storeu_ps(dstPtrRow, pDst);
+                        srcPtr1Row += 8;
+                        dstPtrRow += 8;
+                    }
+                    for (; vectorLoopCount < width; vectorLoopCount++)
+                        *dstPtrRow++ = (*srcPtr1Row++) + additionFactor;
+                }
+            }
+        }
+    }
+    return status;
+}
+
+static vx_status VX_CALLBACK initializeTensorAddTensor(vx_node node, const vx_reference *parameters, vx_uint32 num) {
+    TensorAddTensorLocalData *data = new TensorAddTensorLocalData;
+    memset(data, 0, sizeof(TensorAddTensorLocalData));
+    STATUS_ERROR_CHECK(vxCopyScalar((vx_scalar)parameters[6], &data->deviceType, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxReadScalarValue((vx_scalar)parameters[5], &data->nbatchSize));
```

Comment:
nbatchSize not required, see how to remove and use the n from tensor dims, don't get it as separate parameter in OpenVX